rules/mapping.go

```diff
@@ -85,9 +85,26 @@ func newMappingRuleSnapshotFromFields(
 		name:         name,
 		tombstoned:   tombstoned,
 		cutoverNanos: cutoverNanos,
-		policies:     policies,
-		rawFilters:   tagFilters,
 		filter:       filter,
+		rawFilters:   tagFilters,
+		policies:     policies,
+	}
+}
+
+func (mrs mappingRuleSnapshot) clone() mappingRuleSnapshot {
+	filters := make(map[string]string, len(mrs.rawFilters))
+	for k, v := range mrs.rawFilters {
+		filters[k] = v
+	}
+	policies := make([]policy.Policy, len(mrs.policies))
+	copy(policies, mrs.policies)
+	return mappingRuleSnapshot{
+		name:         mrs.name,
+		tombstoned:   mrs.tombstoned,
+		cutoverNanos: mrs.cutoverNanos,
+		filter:       mrs.filter,
```

Comment:
Are the filters always immutable? Even so, I think it'd be better to add a `Clone` method to the filter interface and have immutable filters just return themselves (possibly with the help with a `newImmutableFilter` ctor). This way if in the future a stateful filter is added, we don't have to change the logic here.